sdk/core/azure-core/test/ut/transport_adapter.cpp

```diff
@@ -439,4 +439,52 @@ namespace Azure { namespace Core { namespace Test {
     CheckBodyFromBuffer(*response, expectedResponseBodySize);
   }
 
+  TEST_F(TransportAdapter, cancelTransferUpload)
+  {
+    Azure::Core::Http::Url host("http://httpbin.org/put");
+    Azure::Core::Context cancelThis;
+
+    auto threadRoutine = [host, cancelThis]() {
+      // Start a big upload and expect it to throw cancelation
+      std::vector<uint8_t> bigBuffer(1024 * 1024 * 200, 'x'); // upload 200 Mb
+      auto stream = Azure::Core::Http::MemoryBodyStream(bigBuffer);
+      auto request = Azure::Core::Http::Request(Azure::Core::Http::HttpMethod::Put, host, &stream);
+
+      // Request will be canceled from main thread throwing the exception
+      EXPECT_THROW(pipeline.Send(cancelThis, request), Azure::Core::RequestCanceledException);
+    };
+
+    // Start request
+    std::thread t1(threadRoutine);
+
+    // Wait 100 ms so we know upload has started
+    std::this_thread::sleep_for(std::chrono::milliseconds(100));
+
+    cancelThis.Cancel();
+    t1.join();
+  }
+
+  TEST_F(TransportAdapter, cancelTransferDownload)
+  {
+    // public big blob (321MB)
+    Azure::Core::Http::Url host("https://vivazqu.blob.core.windows.net/bigfile/321MB");
```

Comment:
Given this is for testing we should put this into our test resources subscription so I went ahead and created a container and uploaded your file there. Please switch to using `https://bigtestfiles.blob.core.windows.net/cpptestfiles/321MB`

Reply:
thank you!

Reply:
@weshaggard Is access to the blob limited?  Given this string is in the repo others will see it.  I wouldn't want someone to update the blob with malicious content.

Reply:
The blob is publicly readable but it is restricted to our team for updating.